src/sagemaker_training/mpi.py

```diff
@@ -341,6 +341,18 @@ def _create_command(self):
             command.extend(["-x", "FI_PROVIDER=efa"])
             # Use simple protocol to handle the out-of-order data delivery from EFA
             command.extend(["-x", "NCCL_PROTO=simple"])
+            # Further set NCCL_ALGO for modelparallel job to avoid hang, starting from torch2.0.0
+            env = environment.Environment()
+            if env.is_modelparallel_enabled:
+                try:
+                    import torch
+
+                    if torch.__version__.split(".")[0] != "1":
+                        command.extend(["-x", "NCCL_ALGO=ring"])
+                except ImportError:
```

Comment:
sometimes if we import a framework, we will see exception other than ImportError like permission error, how do we handle unknown errors caused by importing?(this is part of the reason why we are not suggesting importing framework directly)

Reply:
Updated with changes in SMPTTT